models/quix/optimism/quix_v2_optimism_events.sql

```diff
@@ -0,0 +1,231 @@
+    ,(tr.value / er.amount_raw * 100) as royalty_fee_percentage
+    ,case when tr.value is not null then tr.to end as royalty_fee_receive_address
+    ,case when tr.value is not null
+        then case when (erc20.contract_address = '0x0000000000000000000000000000000000000000' or erc20.contract_address is null) 
+            then 'ETH' else t1.symbol end
+        end as royalty_fee_currency_symbol
+    ,'quix' || '-' || er.tx_hash || '-' || er.token_id || '-' || er.seller || '-' || coalesce(erct2.evt_index::string, '') as unique_trade_id   
+from events_raw as er 
+join {{ source('optimism','transactions') }} as tx 
+    on er.tx_hash = tx.hash
+    and er.block_number = tx.block_number
+    {% if not is_incremental() %}
+    -- smallest block number for source tables above
+    and tx.block_number > 2753613
+    {% endif %}
+    {% if is_incremental() %}
+    and tx.block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+left join {{ ref('nft_aggregators') }} as agg
+    on agg.contract_address = tx.to 
+    and agg.blockchain = 'optimism'
+left join {{ ref('tokens_nft') }} n
+    on n.contract_address = er.nft_contract_address 
+    and n.blockchain = 'optimism'
+left join {{ source('erc721_optimism','evt_transfer') }} as erct2 
+    on erct2.evt_block_time=er.block_time
+    and er.nft_contract_address=erct2.contract_address
+    and erct2.evt_tx_hash=er.tx_hash
+    and erct2.tokenId=er.token_id
+    and erct2.from=er.buyer
+    {% if not is_incremental() %}
+    -- smallest block number for source tables above
+    and erct2.evt_block_number > 2753613
+    {% endif %}
+    {% if is_incremental() %}
+    and erct2.evt_block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+left join {{ source('erc20_optimism','evt_transfer') }} as erc20 
+    on erc20.evt_block_time=er.block_time
+    and erc20.evt_tx_hash=er.tx_hash
+    and erc20.to=er.seller
+    {% if not is_incremental() %}
+    -- smallest block number for source tables above
+    and erc20.evt_block_number > 2753613
+    {% endif %}
+    {% if is_incremental() %}
+    and erc20.evt_block_time >= date_trunc("day", now() - interval '1 week')
+    {% endif %}
+left join {{ ref('tokens_erc20') }} as t1
+    on t1.contract_address =
+        case when (erc20.contract_address = '0x0000000000000000000000000000000000000000' or erc20.contract_address is null)
+        then '0xdeaddeaddeaddeaddeaddeaddeaddeaddead0000'
+        else erc20.contract_address
+        end 
+    and t1.blockchain = 'optimism'
+    left join {{ source('prices', 'usd') }} as p1
+    on p1.contract_address =
+        case when (erc20.contract_address = '0x0000000000000000000000000000000000000000' or erc20.contract_address is null)
+        then '0xdeaddeaddeaddeaddeaddeaddeaddeaddead0000'
+        else erc20.contract_address
+        end
+    and p1.minute = date_trunc('minute', er.block_time)
+    and p1.blockchain = 'optimism'
+    {% if is_incremental() %}
```

Comment:
Since we're joining on minute above, does this add any performance gains? (cc @jeff-dude)

Reply:
yeah, we have been adding time filter on prices too. it helps.